src/core/stream_set.c

```diff
@@ -623,18 +623,24 @@ QuicStreamSetGetStreamForPeer(
                     FrameIn0Rtt,                    // Opened0Rtt
                     &Stream);
             if (QUIC_FAILED(Status)) {
+                *ProtocolViolation = TRUE;
```

Comment:
I could rename this something more generic like "FatalError" to indicate that the connection has been killed.